src/storage/orderbook.hpp

```diff
@@ -0,0 +1,79 @@
+#include "common.hpp"
+
+#include <glaze/glaze.hpp>
+#include <hiredis/hiredis.h>
+
+namespace raccoon {
+namespace storage {
+
+struct product_tracker {
+    std::unordered_map<double, double> bids;
+    std::unordered_map<double, double> asks;
+};
+
+struct OrderbookSnapshot {
+    std::string type = "snapshot";
+    std::string time;
+    std::string product_id;
+    std::vector<std::tuple<std::string, std::string>> asks;
+    std::vector<std::tuple<std::string, std::string>> bids;
+};
+
+struct OrderbookUpdate {
+    std::string type = "l2update";
+    std::string time;
+    std::string product_id;
+    std::vector<std::tuple<std::string, std::string, std::string>> changes;
+};
+
+class OrderbookProcessor {
+private:
+    std::unordered_map<std::string, product_tracker> orderbook_;
+
+public:
+    void process_incoming_snapshot_(const OrderbookSnapshot& newOb);
+    void process_incoming_update_(const OrderbookUpdate& newUpdate);
+    void ob_to_redis_(redisContext* redis, const std::string& product_id);
```

Comment:
```suggestion
    void process_incoming_snapshot(const OrderbookSnapshot& newOb);
    void process_incoming_update(const OrderbookUpdate& newUpdate);
    void ob_to_redis(redisContext* redis, const std::string& product_id);
```

Underscore only for private methods